src/custom/state/application/hooks.ts

```diff
@@ -42,3 +43,22 @@ export function useAddPopup(): (content: PopupContent, key?: string, removeAfter
     [dispatch]
   )
 }
+
+// Set pop up closed if it has not been closed and not fulfill a condition such as not pending tx
+export function useCloseFollowTxPopupIfNot(fulfillsCondition: boolean) {
+  const setShowFollowTxPopup = useUpdateAtom(handleFollowPendingTxPopupAtom)
+  const showingPopup = selectAtom(followPendingTxPopupAtom, ({ showPopup }) => showPopup)
```

Comment:
When the popup is displayed and there was no action to **close it**, its status remains **showing**. I use this hook to follow when the status changes in `Web3StatusInner` to have no pending transactions and set the status back to not visible.

Reply:
Right, but the update will depend on the pending status, not whether it's visible, no?

Reply:
I had misunderstood, because you were referring to line [50](https://github.com/cowprotocol/cowswap/blob/745-show-pop-up/src/custom/state/application/hooks.ts#L50) not the hook on [48](https://github.com/cowprotocol/cowswap/blob/745-show-pop-up/src/custom/state/application/hooks.ts#L48).

I have reused `showFollowTxPopup`